manifestoes/extensions.rst

```diff
@@ -0,0 +1,1675 @@
+We'll use this definition going forward.
+
+Why do we have warnings at all? If something is wrong with a user program,
+shouldn't we just stop compilation? It is tempting to say that a warning
+is a *recoverable* problem with a user program. That is, an error is a problem
+where the compiler can't make sense of the user's program, while a warning
+doesn't prevent the compiler from understanding the user program -- it's just
+identifying a potential problem.
+
+Yet that doesn't quite hold up under scrutiny:
+with ``-fdefer-type-errors``, GHC is perfectly capable of making sense of a
+program with type errors. With ``-XNoDeriveFunctor``, GHC is similarly capable
+of making sense of a program with ``deriving Functor``. Yet in both of these
+scenarios (and many more), GHC issues an error.
+
+I thus argue that the line between warnings and (some) errors is blurry. We
+can imagine a spectrum of code problems going from spelling of identifiers
+through the kinds of problems identified by Hlint through GHC's warnings through
+features enabled by GHC extensions through features enabled by flags like ``-fdefer-type-errors``
+through errors GHC simply cannot recover from (like parser errors). Embodied
+in the current way these potential problems are distributed between different
+reporting mechanisms are value judgments on what the GHC (and other tool) developers
+think about the problems at hand. It is thus reasonable to expect people to differ
+on these points, as with any value judgments. My point here is simply calling
+something an "error" or a "warning" or some other problem is not very informative.
+Accordingly, we should strive for a *uniform treatment* of these potential problems;
+doing so would simplify Haskell's user interface.
+
+Let's also reflect on why we have warnings at all. I claim the usefulness of
+warnings depends on what our user is trying to accomplish. Here are some
+possibilities:
+
+1. The user is actively developing the code in question.
+2. The user is compiling the code in question in order to use the compilation
+   output (as a library or an application); the user is not a developer of the
+   code.
+3. The user is running CI and/or publishing their work.
+
+In scenario (1), warnings are great. They give the user information they might
+use to improve their code. Indeed warnings are better than errors for them, because
+perhaps their focus at the moment is on a different part of the code than where
+the warning arose, or perhaps the user is experimenting and is happy for e.g. an
+import to be unused.
+
+In scenario (2), warnings are mostly useless. The user doesn't care about the
+code being compiled, and doesn't need to know that it has a few unused variables.
+The only reason that warnings aren't completely useless here is that conscientious
+users might reach out to library authors to tell them that their code is warning.
+
+In scenario (3), warnings are actively harmful. When going through final checks
```

Comment:
I found this sentence confusing - the previous paragraphs are talking about how good the warning _mechanism_ is, here I think you're talking about the presence of warnings. I think it would be more consistent to say
```suggestion
In scenario (3), warnings are critical. When going through final checks
```

Reply:
Interesting! I was intending a different emphasis here than came across. I will revise.